src/analyzer/protocol/ssh/ssh-protocol.pac

```diff
@@ -37,6 +37,7 @@ type SSH_Key_Exchange(is_orig: bool) = record {
 	key_ex: case $context.connection.get_version() of {
 		SSH1 -> ssh1_msg : SSH1_Key_Exchange(is_orig, packet_length);
 		SSH2 -> ssh2_msg : SSH2_Key_Exchange(is_orig, packet_length);
+		default -> terminate : bytestring &restofdata &transient;
```

Comment:
Chatting with @J-Gras: He observes this causing the SSH analyzer to not produce violations, when testing with HTTP traffic on port 22, for example.

For 7.0 we're aiming to revert this.

> This fix is very helpful in some operational environments, where analyzer.log can wind up being bigger than ssh.log (!).

It's true that analyzer.log can produce a lot of output and might be too noisy for operational environments. Violations should only be logged until the SSH analyzer is disabled via dpd.zeek. To reduce the volume of the log, there's other mechanisms. These range from disabling the log completely to ignoring the SSH analyzer specifically. Also, log policy hooks could probably help for rate limiting, too.

After further discussion and pondering, prior to 3769ed6c6687e5ecf93c3668bebe3c3f7da3d0fb, `get_version()` would return what had been seen from the client.

One idea for a different fix of the half-duplex case would be to change `get_version()` to return either client or server version if `version_` wasn't determined yet and weird if it's the special 1.99 case.

Reply:
I agree that the original fix is wrong and should be reverted. I like the idea you float about the `get_version()` fix. Another that would work (but I don't know how to do in BinPAC) is have the `default` inform DPD about an analyzer mismatch and detach.